newIDE/app/src/MainFrame/Preferences/PreferencesProvider.js

```diff
@@ -268,17 +270,45 @@ export default class PreferencesProvider extends React.Component<Props, State> {
     }
   }
 
-  _persistValuesToLocalStorage(preferences: Preferences) {
+  _savePreferencesValues(values: PreferencesValues) {
     try {
-      localStorage.setItem(
-        LocalStorageItem,
-        JSON.stringify(preferences.values)
-      );
+      localStorage.setItem(LocalStorageItem, JSON.stringify(values));
     } catch (e) {
       console.warn('Unable to persist preferences', e);
     }
+  }
+
+  _loadLatestPath(project: gdProject, kind: ResourceKind) {
+    try {
+      const projectName = project.getName();
+      const values = this._loadPreferencesValues();
+      if (values) {
+        const curProjectPaths = values.latestPath[projectName];
+        if (curProjectPaths) {
+          console.log(curProjectPaths, curProjectPaths[kind]);
+          return curProjectPaths[kind];
+        }
+      }
+    } catch (e) {
+      console.warn('Unable to load latest path', e);
+    }
+  }
 
-    return preferences;
+  _saveLatestPath(project: gdProject, kind: ResourceKind, path: string) {
+    try {
+      const projectName = project.getName();
+      let values = this._loadPreferencesValues();
```

Comment:
Don't load the preferences. They are already in this.state

Instead, you must update the state - that's why we're using a React context.

Saving the preferences is just a side effect - but we could even not do this, or postpone it to save later, or by a worker, or in 2 hours... the most important is to update the state. See how it's done in other `_setXXX` functions, like `_setAutoDownloadUpdates `

Reply:
Oh! I forgot the fact that the state contains the values. I will update it accordingly.